src/dos/dos.cpp

```diff
@@ -43,120 +45,152 @@ uint16_t countryNo = 0;
+	{ COUNTRY::Tadjikistan    , DATE_DMY , SEP_SLASH  , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // tg_TJ
+	{ COUNTRY::Turkmenistan   , DATE_DMY , SEP_PERIOD , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // tk_TM
+	{ COUNTRY::Azerbaijan     , DATE_DMY , SEP_PERIOD , TIME_24H , SEP_COLON  , SEP_PERIOD , SEP_COMMA  }, // az_AZ
+	{ COUNTRY::Georgia        , DATE_DMY , SEP_PERIOD , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // ka_GE
+	{ COUNTRY::Kyrgyzstan     , DATE_DMY , SEP_SLASH  , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // ky_KG
+	{ COUNTRY::Uzbekistan     , DATE_DMY , SEP_SLASH  , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // uz_UZ
+	//                        | Date fmt | Date separ | Time fmt | Time separ | 1000 separ | Dec separ  |
 };
 
 void DOS_SetCountry(uint16_t country_number)
 {
 	if (dos.tables.country == NULL)
 		return;
 
-	Bit8u *p;
+	// Select country information structure
+	CountryInfo country_info = COUNTRY_INFO[0];
+	for (auto &current : COUNTRY_INFO) {
+		if (current.country_number != country_number) continue;
+
+		country_info = current;
+		break;
+	}
+
+	uint8_t *p;
 
 	// Date format
-	p = dos.tables.country;
-	if (contains(countries_date_format_mmddyyyy, country_number))
-		*p = 0; // MM-DD-YYYY
-	else if (contains(countries_date_format_yyyymmdd, country_number))
-		*p = 2; // YYYY-MM-DD
-	else
-		*p = 1; // DD-MM-YYYY
+	p  = dos.tables.country;
+	*p = country_info.date_format;
 
 	// Date separation character
-	p = dos.tables.country + 11;
-	if (contains(countries_date_separator_slash, country_number))
-		*p = 0x2f; // Forward-slash (/)
-	else if (contains(countries_date_separator_period, country_number))
-		*p = 0x2e; // Period (.)
-	else
-		*p = 0x2d; // Dash (-)
+	p  = dos.tables.country + 11;
+	*p = country_info.date_separator;
 
 	// 12-hour or 24-hour clock
-	p = dos.tables.country + 17;
-	if (contains(countries_time_12hour_clock, country_number))
-		*p = 0; // 12-hour clock
-	else
-		*p = 1; // 24-hour clock
+	p  = dos.tables.country + 17;
+	*p = country_info.time_format;
 
 	// Time separation character
-	p = dos.tables.country + 13;
-	if (contains(countries_time_separator_comma, country_number))
-		*p = 0x2c; // Comma (,)
-	else if (contains(countries_time_separator_period, country_number))
-		*p = 0x2e; // Period (.)
-	else
-		*p = 0x3a; // Column (:)
+	p  = dos.tables.country + 13;
+	*p = country_info.time_separator;
 
 	// Thousand and decimal separators
-	p = dos.tables.country + 7;
-	if (contains(countries_decimal_separator_comma, country_number)) {
-		*p = 0x2c; // Comma (,)
-		*(p + 2) = 0x2e; // Period (.)
-	} else if (contains(countries_decimal_separator_period, country_number)) {
-		*p = 0x2e; // Period (.)
-		*(p + 2) = 0x2c; // Comma (,)
-	} else {
-		*p = 0x20; // Space ( )
-		*(p + 2) = 0x2c; // Comma (,)
-	}
+	p  = dos.tables.country + 7;
+	*p = country_info.thousands_separator;
+	p  = dos.tables.country + 9;
+	*p = country_info.decimal_separator;
```

Comment:
We've got a couple places where these pesky magic country-attribute offsets are used, and different techniques are used in both places too (array index vs. pointer arithmetic).  

![2022-03-29_17-38](https://user-images.githubusercontent.com/1557255/160757947-02638aad-a4de-4df2-a5c9-15a7d02e744c.png)

Ripe for a bit more cleanup in both (maybe make the offsets constexpr's too?) You're call; appreciate either approach :+1: